include/dynd/bytes.hpp

```diff
@@ -89,14 +89,22 @@ class DYND_API bytes {
 
   const bytes operator+(const bytes& rhs)
   {
-      bytes result;
+    bytes result;
 
-      result.resize(size() + rhs.size());
+    result.resize(size() + rhs.size());
 
-      DYND_MEMCPY(result.begin(), begin(), size());
-      DYND_MEMCPY(result.begin() + size(), rhs.begin(), rhs.size());
+    DYND_MEMCPY(result.begin(), begin(), size());
+    DYND_MEMCPY(result.begin() + size(), rhs.begin(), rhs.size());
 
-      return result;
+    return result;
+  }
+
+  bytes &operator+=(const bytes& rhs)
+  {
+    size_t orig_size = size();
+    resize(size() + rhs.size());
+    DYND_MEMCPY(begin() + orig_size, rhs.begin(), rhs.size());
+    return *this;
```

Comment:
It's pretty close, but I think it needs API to distinguish the exponential growth mode, so that can be used appropriately. The ABI is quite a bit different, that's something that needs to be solidified before connecting it with Numba or another JIT. I've made https://github.com/libdynd/libdynd/pull/907 to give one possible way the exponential growth can be expressed, I can turn that into a code PR later if that API looks alright.